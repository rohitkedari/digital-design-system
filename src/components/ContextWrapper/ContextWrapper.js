/**
 * context wrapper component
 */
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeContext, ResponsiveContextProvider } from 'contexts';
import { deepMerge, GlobalStyle, loadFonts } from 'utils';
import { baseTheme } from 'themes/base';

const ContextWrapper = ({ children, theme }) => {
    const nextTheme = deepMerge(baseTheme, theme);
    loadFonts(nextTheme.global.font.url);
    return (
        <ThemeContext.Provider value={nextTheme}>
            <ResponsiveContextProvider
                breakpoints={nextTheme.global.breakpoints}
            >
                <GlobalStyle />
                {children}
            </ResponsiveContextProvider>
        </ThemeContext.Provider>
    );
};
ContextWrapper.defaultProps = {
    theme: null,
};
ContextWrapper.propTypes = {
    children: PropTypes.node.isRequired,
    theme: PropTypes.shape({}),
};
export { ContextWrapper };
