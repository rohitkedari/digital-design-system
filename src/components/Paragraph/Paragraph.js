import React from 'react';

const Paragraph = () => {
	return <p />;
};

export { Paragraph };
