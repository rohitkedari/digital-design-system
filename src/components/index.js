/* GENERATOR: Assemble all components for export */
export * from './Modal';
export * from './Select';
export * from './ContextWrapper';
export * from './Heading';
export * from './Paragraph';
export * from './Icon';
