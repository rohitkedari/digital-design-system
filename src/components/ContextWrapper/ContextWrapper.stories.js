import React, { useContext } from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import { ThemeContext, ResponsiveContext } from 'contexts';
import { withA11y } from '@storybook/addon-a11y';
import { ContextWrapper } from './ContextWrapper';

const StyledContent = styled('div')`
	color: ${(props) => props.color || props.theme.global.colors.brand};
`;
storiesOf('components|ContextWrapper', module)
	.addParameters({ component: ContextWrapper })
	.addDecorator(withA11y)
	.add('Default Context', () => {
		const Content = () => {
			const { viewport } = useContext(ResponsiveContext);
			const theme = useContext(ThemeContext);
			return (
				<div>
					{`This is a ${viewport} screen`}
					<StyledContent>Text with brand color</StyledContent>
					<StyledContent color={theme.global.colors.status_ok}>
						Text with success color
					</StyledContent>
					<StyledContent color={theme.global.colors.status_error}>
						Text with error color
					</StyledContent>
					<StyledContent color={theme.global.colors.status_warning}>
						Text with warning color
					</StyledContent>
				</div>
			);
		};
		return (
			<ContextWrapper>
				<Content />
			</ContextWrapper>
		);
	});
