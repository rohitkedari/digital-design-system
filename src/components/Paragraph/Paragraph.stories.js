import React from 'react';
import { storiesOf } from '@storybook/react';
import { withA11y } from '@storybook/addon-a11y';
import { Paragraph } from './Paragraph';

storiesOf('components|Paragraph', module)
	.addParameters({ component: Paragraph, jest: ['Paragraph.test.js'] })
	.addDecorator(withA11y)
	.add('Default Paragraph', () => <Paragraph>Default paragraph</Paragraph>);
