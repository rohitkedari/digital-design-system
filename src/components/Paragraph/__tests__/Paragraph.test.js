import React from 'react';
import renderer from 'react-test-renderer';
import { ContextWrapper } from 'components/ContextWrapper';
import { Paragraph } from '../Paragraph';

test('Paragraph default renders correctly', () => {
	const tree = renderer
		.create(
			<ContextWrapper>
				<Paragraph>Test Paragraph</Paragraph>
			</ContextWrapper>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
