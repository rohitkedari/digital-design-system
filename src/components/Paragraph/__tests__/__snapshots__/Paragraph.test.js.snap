// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph default renders correctly 1`] = `<p />`;
